app/services/csv_exporter.rb

```diff
@@ -101,4 +130,30 @@ def dequote_and_truncate(text)
   def humanize_boolean(boolean)
     boolean ? 'Yes' : nil
   end
+
+  def num_days_late(kase)
+    if kase.date_draft_compliant.present? && kase.internal_deadline.present?
+      days = (kase.date_draft_compliant - kase.internal_deadline).to_i
+      days > 0 ? days : nil
+    end
+  end
+
+  # Caseworker officer is blank for non-trigger and
+  # always a disclosure specialist for trigger cases.
+  # Catch exception from Case#assigned_disclosure_specialist
+  def casework_officer(kase)
+    return unless kase.workflow == 'trigger'
+
+    kase.assigned_disclosure_specialist.user.full_name rescue nil
```

Comment:
really not sure about these rescue nil lines. Certainly the one in 'draft_in_time' doesn't look right. Why would any of these methods throw an exception, and if they did why would we want to ignore it?

Reply:
In `case/base.rb` the method `assigned_disclosure_specialist` throws an exception if there are no assigning users:

```
  def assigned_disclosure_specialist
    ass = assignments.approving.accepted.detect{ |a| a.team_id == BusinessUnit.dacu_disclosure.id }
    raise 'No assigned disclosure specialist' if ass.nil? || ass.user.nil?
    ass.user
  end
```

In my opinion not being able to generate a report because of a single shoddy record seems unfair.

Reply:
there shouldn't be any shoddy records. If there are, then this code just hides the bugs

Reply:
Good discussion. Ended up creating a 'non bang!' version of the method which returns `nil` when no user is available rather than throw an exception.